cargo-dist/src/backend/ci/github.rs

```diff
@@ -238,6 +322,60 @@ impl GithubCiInfo {
     }
 }
 
+fn vendor_repo(
+    root: &Utf8PathBuf,
+    nwo: &GithubRepoPair,
+    revision: &str,
+    tools: &Tools,
+) -> DistResult<()> {
+    let git = if let Some(tool) = &tools.git {
+        tool.cmd.to_owned()
+    } else {
+        return Err(DistError::ToolMissing {
+            tool: "git".to_owned(),
+        });
+    };
+
+    let url = format!("https://github.com/{nwo}.git");
+    let repo_path = root.join(&nwo.repo);
+    // If we're updating an existing vendored copy, remove it before
+    // cloning a new copy.
+    if repo_path.exists() {
+        std::fs::remove_dir_all(&repo_path)?;
+    }
+
+    let (_tmp_dir, tmp_root) = create_tmp()?;
+
+    info!("Vendoring action {}@{}", nwo, revision);
+    // First we have to clone the repo
+    Cmd::new(&git, "clone repository")
+        .arg("clone")
+        .arg(&url)
+        .arg("--recursive")
+        // We use the separate-git-dir trick here because we don't
+        // actually want to treat this as a git clone: we're just
+        // using git to download for convenience.
+        // Using --separate-git-dir pointed at a temporary directory
+        // is easier than deleting the .git subdir later.
+        .arg("--separate-git-dir")
```

Comment:
i know almost nothing about this.... fascinating

Reply:
Got removed by me swapping over to the tarball, but this *is* a surprisingly useful trick.